docs/proposals/20201019-prometheus-rules-in-observatorium-api.md

```diff
@@ -0,0 +1,271 @@
+
+The downside here is that users may already have 'mixed' rules files, which they would need to split out in order to be accepted into the API.
+
+###### 2. Accept Request With Warning
+
+The other option would be to accept the rules request, store the recording rules and alerting rules alongside one another, but return a warning to saying `Warning: alerting rules are not currently support in Observatorium - please see the roadmap.`
+
+Under this option, we would also have to decide whether alerting rules are loaded into Rule and evaluated or only stored?
+
+###### Alert Decision
+
+My opinion is that the [least surprising](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) option for tenants is for their requests to be explicitly rejected with a clear error message.
+
+While this may impose some file splitting costs on the user's side, we avoid ambiguity such as valid questions such as:
+* Did all of my rules get stored or just some?
+* Are my alerting rules being evaluated or just the recording rules?
+
+Finally, warnings are easy to ignore especially if users are using automated process to interact with the API.
+
+#### Storage layer
+
+We will use a persistent storage mechanism that is external to the API deployment (i.e. no local storage). This enables us to satisfy our requirement of horizontally scaling the rule storage backend.
+
+While we have explicitly de-coupled the rule storage backend from the API, we require a concrete implementaion.
+
+We have a number of options for the storage layer backing the rules storage backend service.
+
+##### 1. ETCD
+
+One option would be to use the Kubernetes' control plane as the backing data store for tenant's recording rules.
+
+Pros:
+* Lowest possible overhead. No external storage is required to be orchestrated.
+* ETCD data model well suited for recording rules i.e. YAML.
+* Storage permissions are managed as Kubernetes objects.
+* We could leverage Recording Rules CRDs and get validation for free (?)
+
+Cons:
+* Implicitly ties Observatorium's implementation into a Kubernetes environment. Hard to back out.
+* Large blast radius. With a large number of tenants and recording rules, we could un-intentionally impact the Kubernetes control plane's performance.
+* ETCD has a limit of 1.5MB per key / value pair so we can't store large files (but Prometheus operator seems to do this ok?).
+
+##### 2. Object Storage
+
+Pros:
+* Users of Thanos project are likely to already have object storage configured and ready to use.
+* Recording rule data model is a good fit for object storage i.e. YAML.
+
+Cons:
+* Unclear consistency semantics depending on object storage provider i.e. in the case of lots of writes, who wins?
+* Requires tenants to manage another set of secret credentials.
```

Comment:
Really? I though this storage could be behind our API as in this proposal so we would maintain credentials to storage